test/e2e/e2e_test.go

```diff
@@ -1029,6 +1079,297 @@ var _ = Describe("e2e multiple external gateway update validation", func() {
+		egressIP2[len(egressIP2)-2]++
+		egressIP2[len(egressIP2)-1]++
+
+		var egressIPConfig = fmt.Sprintf(`apiVersion: k8s.ovn.org/v1
+kind: EgressIP
+metadata:
+    name: egressip
+spec:
+    egressIPs:
+    - ` + egressIP1.String() + `
+    - ` + egressIP2.String() + `
+    podSelector:
+        matchLabels:
+            wants: egress
+    namespaceSelector:
+        matchLabels:
+            name: ` + f.Namespace.Name + `
+`)
+		if err := ioutil.WriteFile(egressIPYaml, []byte(egressIPConfig), 0644); err != nil {
+			framework.Failf("Unable to write CRD config to disk: %v", err)
+		}
+		defer func() {
+			if err := os.Remove(egressIPYaml); err != nil {
+				framework.Logf("Unable to remove the CRD config from disk: %v", err)
+			}
+		}()
+
+		framework.Logf("Applying the EgressIP configuration")
+		framework.RunKubectlOrDie("apply", "-f", egressIPYaml)
+
+		targetExternalContainerAndTest := func(verifyIPType, podName string, verifyIPs []string) {
+			framework.RunKubectlOrDie("exec", podName, frameworkNsFlag, "--", "curl", net.JoinHostPort(targetNode.nodeIP, "80"))
+			targetNodeLogs, err := runCommand("docker", "logs", targetNode.name)
+			if err != nil {
+				framework.Failf("failed to inspect logs in test container: %v", err)
+			}
+			targetNodeLogs = strings.TrimSuffix(targetNodeLogs, "\n")
+			logLines := strings.Split(targetNodeLogs, "\n")
+			lastLine := logLines[len(logLines)-1]
+			var found bool
+			for _, verifyIP := range verifyIPs {
+				if strings.Contains(lastLine, verifyIP) {
+					found = true
+				}
+			}
+			if !found {
+				framework.Failf("the test external container did not have any trace of the %s IPs: %s being logged, last logs: %s", verifyIPType, verifyIPs, logLines[len(logLines)-1])
+			}
+		}
+
+		type status struct {
+			node     string
+			egressIP string
+		}
+
+		testStatus := func() []status {
+			nodeStdout, err := framework.RunKubectl("get", "eip", "-o", "jsonpath='{.items[0].status.items[*].node}'")
+			if err != nil {
+				framework.Failf("Error: failed to get the EgressIP object, err: %v", err)
+			}
+			egressIPStdout, err := framework.RunKubectl("get", "eip", "-o", "jsonpath='{.items[0].status.items[*].egressIP}'")
+			if err != nil {
+				framework.Failf("Error: failed to get the EgressIP object, err: %v", err)
+			}
+			statuses := []status{}
+			for _, n := range strings.Split(nodeStdout, " ") {
+				statuses = append(statuses, status{
+					node: n,
+				})
+			}
+			egressIPStdout = strings.Trim(egressIPStdout, "'")
+			for i, e := range strings.Split(egressIPStdout, " ") {
+				statuses[i].egressIP = e
+			}
+			return statuses
+		}
+
+		By("Checking that the status is of length two and both are assigned to different nodes")
+		statuses := testStatus()
+		if eIP := net.ParseIP(statuses[0].egressIP); eIP == nil {
+			framework.Failf("Error: expected to have the first egress IP, got something else: %s", statuses[0].egressIP)
+		}
+		if eIP := net.ParseIP(statuses[1].egressIP); eIP == nil {
```

Comment:
could it ever be the case that you don't get a statuses with length 2? Should we check for that here?

Reply:
It could, if we only have one node for running the e2e tests, but there's already a check for that here: https://github.com/ovn-org/ovn-kubernetes/blob/5f9659b303d705fe44960e40ee648c940fc4fcb3/test/e2e/e2e_test.go#L1128 which guarantees that.

Reply:
Well, in the end I suppose there can be a screw up with the KIND network which would mean the egress IPs cannot be hosted on any available KIND node. So I've added a check that the length is 2